### PR TITLE
Implement feature #1: Load expenses from CSV

### PR DESCRIPTION
**Pull request by saschahuberzh**
This PR solves #1 

* implemented the load function in ExpenseStore class 
   : load a file from the data at the location designated by path
   : split each row by , and change type of some elements like amount
   : store each row information in one Expense instance
   : store each Expense instance into ArrayList<Expense>

* How to test
   : after making a file, load that file and check if all element is located correctly using junit

* Happy_path test
   : make file named expenses.csv and load it

*Edge_case test
   : make file empty and see how the program reacts, as well how it reacts with a problem in the list

### CODE REVIEW

**Comment by hayat711** [comment]
**Group 7** will be reviewing this PR [code implemented by group8  for feature #1 ]

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        // Expense 3e
+        Expense e3 = list.get(2);
+        assertEquals("2025-10-03", e3.getDate());
+        assertEquals("Cara", e3.getPayer());
+        assertEquals(new BigDecimal("90.00"), e3.getAmount());
+        assertEquals("USD", e3.getCurrency());
+        assertEquals(List.of("Alice", "Bob", "Cara"), e3.getParticipants());
+        assertEquals("Groceries", e3.getCategory());
+        assertEquals("Market", e3.getNotes());
+    }
```
Comment (bbirribbarribbo) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: The code is highly repetitive. So I am hard to read it! Can you simplify this code?

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 2]
```diff
@@ -1,15 +1,62 @@
...
+
+        if(expenseArray.length != 7){
+          System.out.println("Skipping row "+row_num+": some elements are missing") ;
+          continue ;
+        }
+
+        String date = expenseArray[0].trim();
+        String payer = expenseArray[1].trim();
+        BigDecimal amount = new BigDecimal(expenseArray[2].trim());
+        String currency = expenseArray[3];
```
Comment (ljseop1030) [inline src/main/java/itm/oss/splitter/ExpenseStore.java]: I think it would be better if you also add **.trim()** to currency too.

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 3]
```diff
@@ -1,15 +1,62 @@
...
   public ArrayList<Expense> load(String path) throws IOException {
     // TODO (Issue 1): parse CSV file into Expense list.
     // Format: date,payer,amount,currency,participants,category,notes
     // participants are semicolon-separated.
-    throw new UnsupportedOperationException("load() not implemented yet");
+
+    // path = "data/expenses.csv"
+    ArrayList<Expense> returnList = new ArrayList<>() ; // the list that we will return
+
+    try (Scanner scanner = new Scanner(Paths.get(path))) {
```
Comment (ljseop1030) [inline src/main/java/itm/oss/splitter/ExpenseStore.java:19]: I suggest that you move the **try-catch block** inside the while loop. This way, if one line causes an error, the program can skip it and continue processing the rest of the file.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        Path csvFile = tempDir.resolve("malformed.csv");
+        Files.write(csvFile, badData.getBytes(StandardCharsets.UTF_8));
+
+        ExpenseStore store = new ExpenseStore();
+        ArrayList<Expense> list = store.load(csvFile.toString());
+
+        assertEquals(1, list.size(), "Ignore malformed file");
+    }
+
+}
```
Comment (bbirribbarribbo) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: I want to add edge case. Especially, I wonder you try to load nonexistent test! Because it is crucial o ensure the system is robust and handles the code gracefully, preventing unexpected crashes.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+        // Expense 3e
+        Expense e3 = list.get(2);
+        assertEquals("2025-10-03", e3.getDate());
+        assertEquals("Cara", e3.getPayer());
+        assertEquals(new BigDecimal("90.00"), e3.getAmount());
+        assertEquals("USD", e3.getCurrency());
+        assertEquals(List.of("Alice", "Bob", "Cara"), e3.getParticipants());
+        assertEquals("Groceries", e3.getCategory());
+        assertEquals("Market", e3.getNotes());
```
Comment (bbirribbarribbo) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: I think it is highly repetitive. So I am hard to read it. Can you simply this code??
```suggestion
assertExpenseEquals(
        list.get(0),
        "2025-10-01", "Alice", "60.00", "USD", 
        List.of("Alice", "Bob", "Cara"), "Food", "Pizza night"
    );

    assertExpenseEquals(
        list.get(1),
        "2025-10-02", "Bob", "120.00", "USD", 
        List.of("Alice", "Bob", "Cara"), "Transport", "Taxi + subway"
    );

    assertExpenseEquals(
        list.get(2),
        "2025-10-03", "Cara", "90.00", "USD", 
        List.of("Alice", "Bob", "Cara"), "Groceries", "Market"
    );
```

**Review by ljseop1030 — CHANGES_REQUESTED** [review]
Left 2 reviews that can enhance consistency and exception handling.

**Review by bbirribbarribbo — CHANGES_REQUESTED** [review]
Hello, I am Sihyun.

your code is very nice and clear. However, I'd like you fix your code!

1.Add more Edge test because it is crucial to check your code handling well.
2.Simplify your happy case because your code is highly repetitive!

Can you change your code? Your code is awesome overall!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class ExpenseStoreTest {
+
+    @TempDir
+    Path tempDir;
+
+    @Test
+    void load_happyPath() throws Exception {
```
Comment (ihh25) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: The method name loadEmptyList on line 72 uses camelCase, but here load_happyPath uses an underscore (_). I recommend to unify the style to camelCase, like loadHappyPath().

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
```
Comment (ihh25) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: The method name `loadEmptyList()` on line 72 uses camelCase, but here `load_happyPath() `uses an underscore (_).
I recommend you to unify the style to camelCase, like `loadHappyPath()`.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+        ExpenseStore store = new ExpenseStore();
+        ArrayList<Expense> list = store.load(emptyCsv.toString());
+
+        assertEquals(0, list.size(), "No expenses because the CSV file is empty");
+    }
+
+    //problem with the list
+    @Test
+    void loadInvalidLine() throws Exception {
```
Comment (ihh25) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java:84]: The current tests check cases where 'the file exists but the content is empty' or 'the content is partially incorrect'.
The code would be safer if you added a test case to check how the `load()` method behaves when 'the file itself does not exist'.

**Review by ihh25 — COMMENTED** [review]
Great work.👍 This is excellent code with high accuracy! 
I'd like to suggest two modifications.

1. Please unify the method name style. 
2. Please add a test for calling load() with a non-existent file path.

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 9]
```diff
@@ -1,15 +1,79 @@
...
+          System.out.println("Error : Cannot change the String into BigDecimal, so skip row "+row_num);
+        }catch(IllegalArgumentException e){
+          System.out.println("Error : The given argument does not match the format, so skip row "+row_num);
+        }catch(Exception e){
+          System.out.println("Error: " + e.getMessage());
+        }finally{
+          row_num++ ;
+        }
+
+      } // the end of while loop
```
Comment (ihh25) [inline src/main/java/itm/oss/splitter/ExpenseStore.java:70]: Your while loop in load now does two jobs.
It reads lines from the file, and It parses the string to create an Expense object.
If we separate these two jobs, the code will be much cleaner and easier to test!!

**Review by ihh25 — COMMENTED** [review]
Hi, I looked at your load method.😃Good work!
I'd like to recommend one structural change.

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 2 above)
Comment (saschahuberzh) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Makes sense. Solved like suggested.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 4 above)
Comment (sengA7) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: fixed: edge case for a file who doesn't exist

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 3 above)
Comment (MHJeong730) [reply src/main/java/itm/oss/splitter/ExpenseStore.java:19]: I put try-catch block in reading the file because I wanted to detect errors like the file is broken or missing, so I think this try- catch is necessary. Thus, I added new try-catch block that handles the error in the error of each row. Thank you for pointing out this issue!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 8 above)
Comment (MHJeong730) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java:84]: We added new test function loadFileNotFound() to test loading nonexisting file. Thank you for pointing out this issue!

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 9 above)
Comment (MHJeong730) [reply src/main/java/itm/oss/splitter/ExpenseStore.java:70]: I tried to separate them into different loop, but if doing so, the code reads all lines with first loop and parse the last line to Expense object n times. so I think we can maintain the status quo. Thank you for reviewing!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 4 above)
Comment (MHJeong730) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: We added new test function loadFileNotFound() to test loading nonexisting file. Thank you for pointing out this issue!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 5 above)
Comment (MHJeong730) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: I implemented the assertexpenseEquals() method and simplified the code. Thank you for pointing out this issue!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 7 above)
Comment (MHJeong730) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: We unified the naming convention. Thank you for pointing out this issue!

**Review by MHJeong730 — APPROVED** [review]
Summary of the changes that we have made :

1. Unified the naming style of methods.
2. Added trim() to currency.
3. Implemented assertExpenseEquals() method to simplify the test function.
4. Added new try-catch block to detect error occurs in one line and skip the line
5. Added loadFileNotFound() method to test loading nonexisting file

I tried to separate the while loop,but it led to unintended consequences and we decided to maintain the status quo.

**Review by bbirribbarribbo — APPROVED** [review]
Looks good to me!! Your code is very clearer and nicer than before!!

**Review by ihh25 — APPROVED** [review]
LGTM! I confirmed the changes you made! Thank you for applying my opinion. 💻👍

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 9 above)
Comment (ihh25) [reply src/main/java/itm/oss/splitter/ExpenseStore.java:70]: Ah, I see. That makes sense if it complicates the parsing logic. Thanks for the explanation! Let's keep it as is.

**Review by ljseop1030 — APPROVED** [review]
Checked your modification. LGTM!
